src/v/kafka/server/handlers/produce.cc

```diff
@@ -509,8 +509,12 @@ produce_topic(produce_ctx& octx, produce_request::topic& topic) {
             continue;
         }
 
-        const auto& kafka_noproduce_topics
+        auto kafka_noproduce_topics
           = config::shard_local_cfg().kafka_noproduce_topics();
+        if (!octx.rctx.authorized_auditor()) {
```

Comment:
do we have a test for this new behavior?

Reply:
Now yes